src/main/java/thecodex6824/thaumicaugmentation/api/impetus/node/prefab/ImpetusNode.java

```diff
@@ -149,219 +149,272 @@ public boolean removeOutput(IImpetusNode output) {
-        
+
         return outputs.add(toConnect);
     }
-    
+
     @Override
     public boolean removeInputLocation(DimensionalBlockPos toRemove) {
         return inputs.remove(toRemove);
     }
-    
+
     @Override
     public boolean removeOutputLocation(DimensionalBlockPos toRemove) {
         return outputs.remove(toRemove);
     }
-    
+
     @Override
     public Set<DimensionalBlockPos> getInputLocations() {
         return inputs;
     }
-    
+
     @Override
     public Set<DimensionalBlockPos> getOutputLocations() {
         return outputs;
     }
-    
+
     @Override
     public Set<IImpetusNode> getInputs() {
         return inputs.stream().map(loc -> graph.findNodeByPosition(loc)).filter(Objects::nonNull).collect(Collectors.toSet());
     }
-    
+
     @Override
     public Set<IImpetusNode> getOutputs() {
         return outputs.stream().map(loc -> graph.findNodeByPosition(loc)).filter(Objects::nonNull).collect(Collectors.toSet());
     }
-    
+
     @Override
     public DimensionalBlockPos getLocation() {
         return loc;
     }
-    
+
     @Override
     public void setLocation(DimensionalBlockPos location) {
         if (!loc.isInvalid())
             graph.removeNode(this);
-        
+
         loc = location;
         graph.addNode(this);
     }
-    
+
     @Override
     public Vec3d getBeamEndpoint() {
         return new Vec3d(loc.getPos().getX() + 0.5, loc.getPos().getY() + 0.5, loc.getPos().getZ() + 0.5);
     }
-    
+
     @Override
     public boolean shouldPhysicalBeamLinkTo(IImpetusNode other) {
         return true;
     }
-    
+
     @Override
     public boolean shouldEnforceBeamLimitsWith(IImpetusNode other) {
         return true;
     }
-    
+
     @Override
     public void unload() {
         graph.removeNode(this);
     }
-    
+
     @Override
     public void destroy() {
         for (IImpetusNode node : graph.getInputs(this))
             removeInput(node);
-            
+
         for (IImpetusNode node : graph.getOutputs(this))
             removeOutput(node);
-        
+
         unload();
     }
-    
-    protected void initServer() {
+
+    @Override
+    public boolean hasUnloadedNodes(World world) {
+        if (!hasUnloadedNodes) {
+            return false;
+        }
+
+        List<DimensionalBlockPos> invalid = new ArrayList<>();
+
+        List<Boolean> checks = new ArrayList<>();
         for (DimensionalBlockPos pos : inputs) {
-            World world = DimensionManager.getWorld(pos.getDimension());
-            if (world != null && world.provider.getDimension() == pos.getDimension() && world.isBlockLoaded(pos.getPos())) {
-                TileEntity te = world.getTileEntity(pos.getPos());
-                if (te != null) {
-                    IImpetusNode possible = te.getCapability(CapabilityImpetusNode.IMPETUS_NODE, null);
-                    if (possible != null)
+            checks.add(validateNodeInput(world, pos, invalid));
```

Comment:
Instead of using a list, you can do some boolean operations to get the result:
```
boolean someNodeInvalid = false;
for (...)
    someNodeInvalid |= !validateNodeInput(...);
```
Negating the validate results makes `true` mean that a node is invalid, and `or`-ing them together will result in `true` if at least one invalid result occurred.

Reply:
I totally forgot that this exists